src/throttler/throttler.ts

```diff
@@ -145,52 +155,16 @@ export abstract class Throttler<T, R> {
     if (!taskData) {
       throw new Error(`taskData.get(${cursorIndex}) does not exist`);
     }
-    const task = taskData.task;
-    const tname = this.taskName(cursorIndex);
-    const t0 = Date.now();
+    const promises = [this.executeTask(taskData, cursorIndex)];
+    if (taskData.timeoutMillis) {
+      promises.push(this.initializeTimeout(taskData, cursorIndex));
+    }
 
     try {
-      const result = await this.handler(task);
-      const dt = Date.now() - t0;
-      if (dt < this.min) {
-        this.min = dt;
-      }
-      if (dt > this.max) {
-        this.max = dt;
-      }
-      this.avg = (this.avg * this.complete + dt) / (this.complete + 1);
-
-      this.success++;
-      this.complete++;
-      this.active--;
-      if (taskData.wait) {
-        taskData.wait.resolve(result);
-      }
-      this.taskDataMap.delete(cursorIndex);
-      this.process();
-    } catch (err) {
-      if (this.retries > 0) {
-        if (taskData.retryCount < this.retries) {
-          taskData.retryCount++;
-          this.retried++;
-          await backoff(taskData.retryCount, this.backoff);
-          logger.debug(`[${this.name}] Retrying task`, tname);
-          return this.handle(cursorIndex);
-        }
-      }
-
-      this.errored++;
-      this.complete++;
-      this.active--;
-      if (taskData.retryCount > 0) {
-        logger.debug(`[${this.name}] Retries exhausted for task ${tname}:`, err);
-      } else {
-        logger.debug(`[${this.name}] Error on task ${tname}:`, err);
-      }
-      if (taskData.wait) {
-        taskData.wait.reject(err);
-      }
-      this.finish(err);
+      const result = await Promise.race(promises);
+      this.onTaskFulfilled(result, taskData, cursorIndex);
+    } catch (error) {
```

Comment:
nit: Convention is to use the name `err`.

Reply:
Done